architecture/builds.adoc

```diff
@@ -44,3 +44,51 @@ STI builds are a replacement for the OpenShift v2-like developer experience. The
 == Custom Build
 Custom builds are the most sophisticated version of builds, allowing developers to define a builder image which is responsible for the entire process of the build. The custom builder image is a plain Docker image within which the author embeds the logic of the desired build process, such as building RPMs or building
 base Docker images.
+
+[#using-docker-credentials-for-pushing-images]
+== Using Docker credentials for pushing images
+
+In case you want to push the output image into private Docker Registry that
+require authentication or Docker Hub, you have to supply the `dockercfg` file
+with valid Docker Registry credentials.
+
+The `.dockercfg` JSON file usually exists in your home directory and it has following
```

Comment:
Yeah, but for user the easiest is to just invoke `docker login` against the repo.